### PR TITLE
Add availability testing for the SFTP container

### LINKED ISSUE #10368: Create SFTP monitor for the SFTP containers in Stag/Prod so we know if the are unresponsive
Issue says it all - we want to know if the SFTP server is not working. Ideally we'd just SSH into it every 10 minutes or so

### PR DESCRIPTION
**Pull request by dkrylovsb**
This PR adds availability testing for the SFTP container. A new function in the admin app does the checking, which then feeds back into Application Insights as an additional availability check. 

Test Steps:
1. Do `terraform apply`. Ensure Application Insights display status for the SFTP container. 

## Changes
- A new admin app function: `confirm_sftp_access.py`.
- A new availability test in App Insights that uses this function to poll SFTP.

### Testing
- [x] Tested locally?
## Linked Issues
- Fixes #10368

### CODE REVIEW

**Comment by dkrylovsb** [comment]
`terraform plan -target="module.function_app"`: 
[tfplan.txt](https://github.com/CDCgov/prime-reportstream/files/13231551/tfplan.txt)

File: operations/app/functions/admin/ConfirmSFTPAccess/confirm_sftp_access.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+from paramiko import SSHClient, WarningPolicy, AuthenticationException
+
+
+def main(req: func.HttpRequest) -> func.HttpResponse:
+    host = os.getenv("SFTP_HOST")
+
+    if host:
+        try:
+            client = SSHClient()
+            client.set_missing_host_key_policy(WarningPolicy())
```
[the remark this answers is not in the document]
Comment (dkrylovsb) [reply operations/app/functions/admin/ConfirmSFTPAccess/confirm_sftp_access.py:12]: We merely ensure we can connect to the SFTP server; there is no authentication or data transfer per se.

**Comment by dkrylovsb** [comment]
```Terraform will perform the following actions:

  # module.application_insights.azurerm_application_insights_web_test.sftp_test will be created
  + resource "azurerm_application_insights_web_test" "sftp_test" {
      + application_insights_id = "/subscriptions/7d1e3999-6577-4cd5-b296-f518e5c8e677/resourceGroups/prime-data-hub-prod/providers/Microsoft.Insights/components/pdhprod-appinsights"
      + configuration           = <<-EOT
            <WebTest Name="" Id="" Enabled="True" CssProjectStructure="" CssIteration="" Timeout="60" WorkItemIds="" xmlns="http://microsoft.com/schemas/VisualStudio/TeamTest/2010" Description="" CredentialUserName="" CredentialPassword="" PreAuthenticate="True" Proxy="default" StopOnError="False" RecordedResultFile="" ResultsLocale="">
                  <Items>
                    <Request Method="GET" Version="1.1" Url="https://pdhprod-admin-functionapp.azurewebsites.net/api/ConfirmSFTPAccess" ThinkTime="0" Timeout="60" ParseDependentRequests="False" FollowRedirects="True" RecordResult="True" Cache="False" ResponseTimeGoal="0" Encoding="utf-8" ExpectedHttpStatusCode="200" ExpectedResponseUrl="" ReportingName="" IgnoreHttpStatusCode="False"/>
                  </Items>
                </WebTest>
        EOT
      + enabled                 = true
      + frequency               = 300
      + geo_locations           = [
          + "us-va-ash-azr",
          + "us-ca-sjc-azr",
          + "us-fl-mia-edge",
        ]
      + id                      = (known after apply)
      + kind                    = "ping"
      + location                = "eastus"
      + name                    = "pdhprod-sftp-health-check"
      + resource_group_name     = "prime-data-hub-prod"
      + retry_enabled           = true
      + synthetic_monitor_id    = (known after apply)
      + tags                    = {
          + "environment"                                                                                                                                                    = "prod"
          + "hidden-link:/subscriptions/7d1e3999-6577-4cd5-b296-f518e5c8e677/resourceGroups/prime-data-hub-prod/providers/Microsoft.Insights/components/pdhprod-appinsights" = "Resource"
        }
      + timeout                 = 60
    }

  # module.function_app.azurerm_function_app.admin will be updated in-place
  ~ resource "azurerm_function_app" "admin" {
      ~ app_settings                    = {
          + "SFTP_HOST"                          = "pdhprod-sftp01.eastus.azurecontainer.io"
            # (12 unchanged elements hidden)
        }
        id                              = "/subscriptions/7d1e3999-6577-4cd5-b296-f518e5c8e677/resourceGroups/prime-data-hub-prod/providers/Microsoft.Web/sites/pdhprod-admin-functionapp"
        name                            = "pdhprod-admin-functionapp"
        tags                            = {
            "environment" = "prod"
            "managed-by"  = "terraform"
        }
        # (21 unchanged attributes hidden)




        # (4 unchanged blocks hidden)
    }

  # module.function_app.null_resource.admin_function_app_publish is tainted, so must be replaced
-/+ resource "null_resource" "admin_function_app_publish" {
      ~ id       = "889974833837219638" -> (known after apply)
      ~ triggers = {
          ~ "input_json"           = "7a618bc43fa8f8a838a1e393a94367e6" -> "476445cf956814fa20f13fef16864f76"
          ~ "publish_code_command" = <<-EOT
                      az functionapp deployment source config-zip --resource-group prime-data-hub-prod --name pdhprod-admin-functionapp --src function-app-admin.zip --build-remote false --timeout 600
            EOT
        }
    }

Plan: 2 to add, 1 to change, 1 to destroy.
╷
```

**Review by jeremy-page — APPROVED** [review]
Would you mind waiting until Monday to apply? Just in case :D

**Pull request merged**